chapters/slang.adoc

```diff
@@ -0,0 +1,744 @@
+* *Leverage imports*: Use imports instead of includes for better dependency management
+* *Separate shader stages*: Keep different shader stages in separate files
+* *Use interfaces*: Define interfaces for common functionality to enable polymorphism
+
+Example of a well-organized Slang shader:
+
+[source,slang]
+----
+module Rendering.PBR;
+
+import Rendering.Common;
+import Rendering.Lighting;
+
+export interface IMaterial {
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal);
+}
+
+export struct PBRMaterial : IMaterial {
+    float3 albedo;
+    float roughness;
+    float metallic;
+
+    float3 evaluateBRDF(float3 viewDir, float3 lightDir, float3 normal) {
+        // PBR implementation
+        return calculatePBR(albedo, roughness, metallic, viewDir, lightDir, normal);
+    }
+}
+
+[shader("fragment")]
+export float4 fragmentMain(VSOutput input) {
+    // Shader implementation
+}
+----
+
+=== Performance Considerations
+
+* *Use generics judiciously*: While powerful, excessive use of generics can increase compile times
+* *Profile interface dispatch*: Test performance impact of interface-based polymorphism
+* *Leverage compile-time evaluation*: Use Slang's metaprogramming to move work to compile time
+* *Consider specialization*: Generate specialized shader variants for performance-critical paths
+
+=== Vulkan-Specific Best Practices
+
+* *Use parameter blocks*: Organize resources into parameter blocks for better organization
+* *Explicit bindings*: Always specify explicit descriptor set and binding indices
+* *Cross-API compatibility*: Use conditional compilation for Vulkan-specific code
+
+Example of Vulkan-specific resource binding:
+
+[source,slang]
+----
+// Define parameter blocks with explicit bindings
+[[vk::binding(0, 0)]]
+ParameterBlock<GlobalResources> globals;
+
+[[vk::binding(0, 1)]]
+ParameterBlock<MaterialResources> material;
+
+// Access resources through parameter blocks
+float4 albedo = material.albedoMap.Sample(material.sampler, uv);
+float4x4 viewProj = globals.viewProj;
+----
+
+== Advanced Topics
+
+=== Cross-Compilation and Portability
+
+Slang excels at cross-API compatibility:
+
+* *Multi-target compilation*: Compile the same shader for multiple APIs
+* *Conditional compilation*: Use `#if` directives for API-specific code
+* *Parameter blocks*: Use parameter blocks for consistent resource binding across APIs
+
+Example of cross-API shader:
+
+[source,slang]
+----
+// Common shader code
+module Shaders.PBR;
+
+// API-specific resource binding
+#if SLANG_VULKAN
+[[vk::binding(0, 0)]]
+#endif
+ParameterBlock<GlobalResources> globals;
+
+// Common shader logic
+float4 calculateLighting(float3 position, float3 normal) {
+    // Implementation that works across APIs
+}
+----
+
+=== Shader Composition and Reuse
+
+Slang's module system enables powerful composition patterns:
+
+* *Interface-based composition*: Define interfaces for pluggable components
```

Comment:
It's odd to lead with
> Slang's module system enables powerful composition patterns:

and then start talking about interfaces and generics, when those can be used without modules as well. They're independent features.